src/pretix/base/services/cross_selling.py

```diff
@@ -0,0 +1,231 @@
+#
+# This file is part of pretix (Community Edition).
+#
+# Copyright (C) 2014-2020 Raphael Michel and contributors
+# Copyright (C) 2020-2021 rami.io GmbH and contributors
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General
+# Public License as published by the Free Software Foundation in version 3 of the License.
+#
+# ADDITIONAL TERMS APPLY: Pursuant to Section 7 of the GNU Affero General Public License, additional terms are
+# applicable granting you additional permissions and placing additional restrictions on your usage of this software.
+# Please refer to the pretix LICENSE file to obtain the full terms applicable to this work. If you did not receive
+# this file, see <https://pretix.eu/about/en/license>.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+# warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License along with this program.  If not, see
+# <https://www.gnu.org/licenses/>.
+#
+
+from collections import defaultdict
+from decimal import Decimal
+from itertools import groupby
+from math import inf
+from typing import List
+
+from django.utils.functional import cached_property
+
+from pretix.base.models import CartPosition, ItemCategory, SalesChannel
+from pretix.presale.views.event import get_grouped_items
+
+
+class DummyCategory:
+    """
+    Used to create fake category objects for displaying the same cross-selling category multiple times,
+    once for each subevent
+    """
+
+    def __init__(self, category: ItemCategory, subevent):
+        self.id = category.id
+        self.name = f"{category.name} ({subevent})"
```

Comment:
I’d love to display the subevent as a formatted HTML-string like so:

![Bildschirmfoto 2024-10-11 um 12 52 04](https://github.com/user-attachments/assets/4d20e80a-1072-43de-ba7f-c22f3a39826d)

Do you have an idea how to best add it? Give the ItemCategory a subevent property and handle it in the HTML-template? In quick tests, I did not get `__html__` to work, but I would not want `__str__` to return HTML.

```suggestion
        self.name = f"{category.name}"
        self.subevent = subevent
```